src/encoded/authentication.py

```diff
@@ -228,41 +249,81 @@ def get_token_info(self, token, request):
         return None
 
 
+def get_jwt_from_auth_header(request):
+    if "Authorization" in request.headers:
+        try:
+            # Ensure this is a JWT token, not basic auth.
+            # Per https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication and
+            # https://tools.ietf.org/html/rfc6750, JWT is introduced by 'bearer', as in
+            #   Authorization: Bearer something.something.something
+            # rather than, for example, the 'basic' key information, which as discussed in
+            # https://tools.ietf.org/html/rfc7617 is base64 encoded and looks like:
+            #   Authorization: Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==
+            # See also https://jwt.io/introduction/ for other info specific to JWT.
+            [ auth_type, auth_data ] = request.headers['Authorization'].strip().split(' ', 1)
```

Comment:
Not urgent, so doesn't have to be done for this PR, but: I'd like to see a separate subroutine for parsing out this stuff. I hate seeing `.strip()` and `.split(...)` used amidst higher level operations. This kind of formula is done differently in different places and is a risk of problem. Some sort of separately tested `parse_authorization` would be good.